unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AudioSource/ECSAudioSourceComponentHandler.cs

```diff
@@ -55,23 +60,27 @@ public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
         public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, ECSAudioSource model)
         {
             bool isSameClip = model.audioClipUrl == this.model?.audioClipUrl;
-            this.scene = scene;
             this.model = model;
             
-            ApplyCurrentModel();
-            
             // If the clip has changed, we need to forget the old clip
-            if(!isSameClip && promiseAudioClip != null)
+            if (!isSameClip && promiseAudioClip != null)
+            {
+                isAudioClipReady = false;
+                if (audioClip != null)
+                    dataStore.sceneWorldObjects.RemoveAudioClip(scene.sceneData.id, audioClip);
                 keeperAudioClip.Forget(promiseAudioClip);
+            }
             
-            promiseAudioClip = new AssetPromise_AudioClip(model.audioClipUrl, scene.contentProvider);
-            promiseAudioClip.OnSuccessEvent += OnAudioClipLoadComplete;
-            promiseAudioClip.OnFailEvent += OnAudioClipLoadFail;
+            ApplyCurrentModel();
 
-            keeperAudioClip.Keep(promiseAudioClip);
-            
-            sceneID.OnChange -= OnCurrentSceneChanged;
-            sceneID.OnChange += OnCurrentSceneChanged;
+            if (!isAudioClipReady && !isSameClip)
+            {
+                promiseAudioClip = new AssetPromise_AudioClip(model.audioClipUrl, scene.contentProvider);
+                promiseAudioClip.OnSuccessEvent += OnAudioClipLoadComplete;
+                promiseAudioClip.OnFailEvent += OnAudioClipLoadFail;
```

Comment:
This two events are not unsubbed anywhere, you should even if it's an AssetProsime, right?

Reply:
There is no need since we are just recreating the asset promise and when you forgot an asset promise it gets removed, anyway I will change just in case